### DISCUSSION TITLE
[Error: ENOENT: no such file or directory, open 'F:\FinalYear\frontend\.next\BUILD_ID']

### DISCUSSION THREAD

**Discussion by AliRaza190** [vercel/next.js#58523] [Help]
### Summary

I am working on a next app. I have tried to rebuild it but id doesn't works and am facing the below error:
PS F:\FinalYear\frontend> npx next start
   ▲ Next.js 14.0.2
   - Local:        http://localhost:3000

[Error: ENOENT: no such file or directory, open 'F:\FinalYear\frontend\.next\BUILD_ID'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'F:\\FinalYear\\frontend\\.next\\BUILD_ID'
}

### Additional information

```js
package.json
`{
  "name": "frontend",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "node server.js",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@stripe/stripe-js": "^1.54.0",
    "antd": "^5.5.2",
    "aws-crt": "^1.18.3",
    "aws-sdk": "^2.1389.0",
    "axios": "^1.4.0",
    "bootstrap": "^5.3.0",
    "express": "^4.18.2",
    "fs": "^0.0.1-security",
    "http-proxy-middleware": "^2.0.6",
    "mongoose": "^7.6.3",
    "next": "^14.0.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-image-file-resizer": "^0.4.8",
    "react-markdown": "^9.0.0",
    "react-player": "^2.12.0",
    "react-toastify": "^9.1.3"
  }
}
`
```

**Comment by prunusnira** [comment]
for me, 'next build' does not generate 'BUILD_ID' file randomly. it happens in AWS CodeBuild and generated .next folder does not have BUILD_ID file, so server deploy fails.
(CodeBuild build component get node 18.18.2 installed and then download files from our Git repository directly)

**Comment by prunusnira** [comment]
after some investigation, we found that it's RAM shortage.
we used 3GB / 2vCPU for AWS CodeBuild, and it caused several ENOMEM error and forced SIGKILL during build.
even we succeed to build, there was no BUILD_ID file

we increased RAM to 7GB model, and problem fixed.

idk which system are you using, but hope it helps

**Comment by Deepsharma001** [comment]
Just Created .env and and start with and run npm run dev

**Reply by abhishek210** [reply]
its work

**Comment by ZakarFin** [comment]
Same issue here and thanks to this info I solved the same issue (server not starting because .next/BUILD_ID missing) for an on-premise Jenkins build by adding NODE_OPTIONS in front of npm run build:
`NODE_OPTIONS=--max_old_space_size=4096 npm run build`. The build log didn't really indicate anything about dying because of memory, but thanks for documenting this here :)

**Comment by aaliboyev** [comment]
In my case the problem has gone when I included package-lock.json in docker container. Error was the same but when running inside docker container.
